bin/reth/src/config.rs

```diff
@@ -29,6 +29,7 @@ impl Config {
         chain_id: u64,
         genesis_hash: H256,
         disable_discovery: bool,
+        nat_resolution_method: NatResolver,
```

Comment:
Wonder if we should move these args to the Config struct? And maybe add some clap definitions inside the Config struct and flatten them for CLI usage inside the Command struct?